### DISCUSSION TITLE
Initialization with the object already grasped

### DISCUSSION THREAD

**Discussion by zichunxx** [google-deepmind/mujoco#1950] [Asking for Help] [answered]
Hi!  

I want to initialize a model with the object being grasped by the gripper so that the arm does not have to re-grasp each time when resetting the environment in reinforcement learning.

I have tried some methods that end up with the object falling off before it was grasped. Is it feasible to realize this? 

Thanks!

**Comment by Balint-H** [comment, marked as answer]
Have you tried to grasp the object, then saving `data->qpos`, then restore it as needed? (Setting `qvel` and `qacc` to zero as well)

**Comment by zichunxx** [comment]
> Have you tried to grasp the object, then saving `data->qpos`, then restore it as needed? (Setting `qvel` and `qacc` to zero as well)

Really appreciate the important advice! I missed the key factor of setting `qacc` to zero and further induced the unstable reset.

Thanks for your kindness!
